app/observers/message_observer.rb

```diff
@@ -91,18 +91,16 @@ def rejected(contract)
   private
 
   def plan_changed_publish_event!(contract)
-    event_class = case contract
-                  when ServiceContract
-                    ServiceContracts::ServiceContractPlanChangedEvent
-                  when Cinstance
-                    Cinstances::CinstancePlanChangedEvent
-                  end
-
-    if event_class.present?
-      event = event_class.create(contract, User.current)
-
-      Rails.application.config.event_store.publish_event(event)
-    end
+    events = case contract
+             when ServiceContract
+               [ServiceContracts::ServiceContractPlanChangedEvent.create(contract, User.current)]
+             when Cinstance
+               [
+                 Cinstances::CinstancePlanChangedEvent.create(contract, User.current),
+                 Cinstances::CinstancePlanChangedForBuyerEvent.create(contract, contract.account.first_admin)
```

Comment:
Why do we need both always? 🤔

Reply:
One sends a notification to the provider, while the other to the developer. I wouldn't join them though. I think it's good that we control them separately and keeps SR of the class.

Reply:
WDYT?

Reply:
https://github.com/3scale/porta/pull/709#discussion_r266365048